core/server/data/importer/importers/data/settings.js

```diff
@@ -54,6 +58,12 @@ class SettingsImporter extends BaseImporter {
 
         _.each(this.dataToImport, function (obj) {
             obj.key = self.legacySettingsKeyValues[obj.key] || obj.key;
+
+            if (obj.key === 'labs' && labsSetting && labsSetting.value) {
```

Comment:
You're totally right! I came at this backwards and ended up making it complex - fixed it now.

Importer code is looking SO MUCH BETTER 🎉  since I last looked 🌟